MastodonSDK/Sources/MastodonCore/Service/PhotoLibraryService.swift

```diff
@@ -130,23 +130,52 @@ extension PhotoLibraryService {
         .eraseToAnyPublisher()
     }
 
-    static func save(imageData: Data) -> AnyPublisher<Void, Error> {
+    static func save(imageData: Data, imageType: PHAssetResourceType) -> AnyPublisher<Void, Error> {
         guard PHPhotoLibrary.authorizationStatus(for: .addOnly) != .denied else {
             return Fail(error: PhotoLibraryError.noPermission).eraseToAnyPublisher()
         }
-
-        return Future<Void, Error> { promise in
-            PHPhotoLibrary.shared().performChanges {
-                PHAssetCreationRequest.forAsset().addResource(with: .photo, data: imageData, options: nil)
-            } completionHandler: { isSuccess, error in
-                if let error = error {
-                    promise(.failure(error))
-                } else {
-                    promise(.success(Void()))
+        
+        if imageType == .video {
+            return Future<Void, Error> { promise in
+                let filename = UUID().uuidString
+                let path = FileManager.default.temporaryDirectory.appendingPathComponent("\(filename)")
+                let url = path.appendingPathExtension("mp4")
+                do {
+                    try imageData.write(to: url)
+                } catch {
+                    print(error)
+                }
```

Comment:
If the file can’t be written, the video will not be able to be added to the library and so this should fail the future immediately, right?